ajax/entitytreesons.php

```diff
@@ -52,57 +52,68 @@
-        foreach ($entities as $entities_id => &$entity) {
-            $entity['key']   = $entities_id;
+        $entitytree  = $default_entity['is_recursive'] ? getTreeForItem('glpi_entities', $default_entity_id) : [$default_entity['id'] => $default_entity];
+        $adapt_tree = static function (&$entities) use (&$adapt_tree, $base_path) {
+            foreach ($entities as $entities_id => &$entity) {
+                $entity['key']   = $entities_id;
 
-            $title = "<a href='$base_path?active_entity={$entities_id}'>{$entity['name']}</a>";
-            $entity['title'] = $title;
-            unset($entity['name']);
+                $title = "<a href='$base_path?active_entity={$entities_id}'>{$entity['name']}</a>";
+                $entity['title'] = $title;
+                unset($entity['name']);
 
-            if (isset($ancestors[$entities_id])) {
-                $entity['expanded'] = 'true';
-            }
-
-            if ($entities_id == $_SESSION['glpiactive_entity']) {
-                $entity['selected'] = 'true';
-            }
+                if (isset($entity['tree']) && count($entity['tree']) > 0) {
+                    $entity['folder'] = true;
 
-            if (isset($entity['tree']) && count($entity['tree']) > 0) {
-                $entity['folder'] = true;
+                    $entity['title'] .= "<a href='$base_path?active_entity={$entities_id}&is_recursive=1'>
+                <i class='fas fa-angle-double-down ms-1' data-bs-toggle='tooltip' data-bs-placement='right' title='" . __('+ sub-entities') . "'></i>
+                </a>";
 
-                $entity['title'] .= "<a href='$base_path?active_entity={$entities_id}&is_recursive=1'>
-               <i class='fas fa-angle-double-down ms-1' data-bs-toggle='tooltip' data-bs-placement='right' title='" . __('+ sub-entities') . "'></i>
-            </a>";
+                    $children = $adapt_tree($entity['tree']);
+                    $entity['children'] = array_values($children);
+                }
 
-                $children = $adapt_tree($entity['tree']);
-                $entity['children'] = array_values($children);
+                unset($entity['tree']);
             }
 
-            unset($entity['tree']);
-        }
+            return $entities;
+        };
+        $adapt_tree($entitytree);
 
-        return $entities;
-    };
-    $adapt_tree($entitytree);
+        $entitiestree = array_merge($entitiestree, $entitytree);
+    }
 
-    $entitiestree = array_merge($entitiestree, $entitytree);
+    $all_entitiestree[$subckey] = $entitiestree;
+    $GLPI_CACHE->set($ckey, $all_entitiestree);
 }
 
-$all_entitiestree[$subckey] = $entitiestree;
-$GLPI_CACHE->set($ckey, $all_entitiestree);
+/* scans the tree to select the active entity */
+$entitiestree = [];
+$entitytree = $all_entitiestree[$subckey];
+$adapt_tree = static function (&$entities) use (&$adapt_tree, $ancestors) {
```

Comment:
You should use another function name. On first reading, I though L110 was a call to function declared on L65.